app/policies/kudo_policy.rb

```diff
@@ -0,0 +1,33 @@
+class KudoPolicy < ApplicationPolicy
+  def create?
+    can_afford?
+  end
+
+  def show?
+    creator?
+  end
+
+  def update?
+    creator? && created_less_than_5_minutes_ago?
+  end
+
+  def destroy?
+    creator? && created_less_than_5_minutes_ago?
+  end
```

Comment:
```suggestion
  def destroy?
    update?
  end
```

If we do `destroy?` using `update?` method, then if we need to change policy in the future, we can change it only in one place.